wikimedia/wikipedia-ios
# Add remote config for Year in Review

**Phabricator:** https://phabricator.wikimedia.org/T376040

### Notes
This PR adds a pulls and persists our new remote config located at https://donate.wikimedia.org/wiki/MediaWiki:AppsFeatureConfig.json (Fetched when running the main scheme) and https://test.wikipedia.org/wiki/MediaWiki:AppsFeatureConfig.json (Fetched when running the staging scheme). Since this felt similar to Developer Settings flags, only remotely stored, I decided to put the fetching and persisting logic into the existing WMFDeveloperSettingsDataController, which currently handles the local flags from the Developer Settings menu.

To follow a bit of our other donation and fundraising patterns, I have one method (this one called `fetchFeatureConfig`) that fetches and persists the feature config. This will be called when the app launches or resumes. I also have another method called `loadFeatureConfig` which attempts to load the feature config from memory or persistence, if it hasn't been very long (I chose four hours). We can call loadFeatureConfig whenever the app needs to reference the feature config.

I also added versioning to this config and am checking for the proper version before decoding the Codable object. This is a similar pattern to our other donation and fundraising configs.

I removed the old config fetching location in MWKDataStore.m and replaced it with a new call to the data controller's `fetchFeatureConfig`. This config was already set up to fetch upon app foreground or fresh launch every three hours.

### Test Steps
1. Uncomment profile view model's Year in Review item.
2. At the beginning of YearInReviewCoordinator start(), add this code:

```
        guard let iosConfig = WMFDeveloperSettingsDataController.shared.loadFeatureConfig()?.ios.first else {
            return
        }
        
        print(iosConfig.yir.isEnabled)
        print(iosConfig.yir.countryCodes)
        print(iosConfig.yir.primaryAppLanguageCodes)
        print(iosConfig.yir.dataPopulationStartDateString)
        print(iosConfig.yir.dataPopulationEndDateString)
        print(iosConfig.yir.personalizedSlides)
```

3. Add a breakpoint to WMFDeveloperSettingsDataController's `fetchFeatureConfig` method.
4. On device, change the device date to yesterday.
5. Fresh install and run on this PR. Confirm `fetchFeatureConfig` breakpoint is hit (meaning it is fetched).
6. Tap into YiR profile item. Confirm config items are printed out.
7. Terminate app and relaunch. Confirm `fetchFeatureConfig` breakpoint is NOT hit. Tap into YiR profile item. Confirm config items are still printed out.
8. Background app, change device date to 3.5 hours later.
9. Foreground app, confirm `fetchFeatureConfig` breakpoint is hit again.
10. Tap into YiR profile item. Confirm config items are printed out.
11. Terminate app, change device date to 1 day later.
12. Run app again, confirm `fetchFeatureConfig` is hit on launch. Tap into YiR profile item. Confirm config items are printed out.

### Screenshots/Videos
